### PR TITLE
FileService_GetFile

### PR DESCRIPTION
**Pull request by lalagvana**
- Added GetFile Command.
- Added methods to Repository and Controller.
- Bind new command in Startup.cs.
- Added tests for Command and Repository.

### CODE REVIEW

File: DigitalOffice/Tests/FileServiceTests/Commands/GetFileCommandTests.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+using LT.DigitalOffice.FileService.Database.Entities;
+using LT.DigitalOffice.FileService.Mappers.Interfaces;
+using LT.DigitalOffice.FileService.Models;
+using LT.DigitalOffice.FileService.Repositories.Interfaces;
+using LT.DigitalOffice.FileServiceUnitTests.UnitTestLibrary;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
```
Comment (MihailKulikov) [inline DigitalOffice/Tests/FileServiceTests/Commands/GetFileCommandTests.cs]: Unused directives

File: DigitalOffice/Tests/FileServiceTests/Commands/GetFileCommandTests.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+namespace LT.DigitalOffice.FileServiceUnitTests.CommandsTests
```
Comment (MihailKulikov) [inline DigitalOffice/Tests/FileServiceTests/Commands/GetFileCommandTests.cs]: Change namespace

File: DigitalOffice/Services/FileService/Repositories/Interfaces/IFileRepository.cs [hunk 3]
```diff
@@ -1,10 +1,12 @@
 ﻿using LT.DigitalOffice.FileService.Database.Entities;
+using LT.DigitalOffice.FileService.Models;
```
Comment (MihailKulikov) [inline DigitalOffice/Services/FileService/Repositories/Interfaces/IFileRepository.cs]: Unused directive

File: DigitalOffice/Tests/FileServiceTests/Repositories/GetFileRepositoryTests.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
+﻿using LT.DigitalOffice.FileService.Database;
+using LT.DigitalOffice.FileService.Database.Entities;
+using LT.DigitalOffice.FileService.Repositories;
+using LT.DigitalOffice.FileService.Repositories.Interfaces;
+using LT.DigitalOffice.FileServiceUnitTests.UnitTestLibrary;
+using Microsoft.EntityFrameworkCore;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Text;
```
Comment (MihailKulikov) [inline DigitalOffice/Tests/FileServiceTests/Repositories/GetFileRepositoryTests.cs]: Unused directives

File: DigitalOffice/Tests/FileServiceTests/Repositories/GetFileRepositoryTests.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+namespace LT.DigitalOffice.FileServiceUnitTests.Repositories
+{
+    public class GetFileRepositoryTests
+    {
+        private FileServiceDbContext dbContext;
+        private IFileRepository repository;
+        private Guid fileId;
```
Comment (MihailKulikov) [inline DigitalOffice/Tests/FileServiceTests/Repositories/GetFileRepositoryTests.cs]: What are the purposes of this field?

**Review by MihailKulikov — APPROVED** [review]

File: DigitalOffice/Tests/FileServiceTests/Repositories/GetFileRepositoryTests.cs (hunk 5 above)
Comment (lalagvana) [reply DigitalOffice/Tests/FileServiceTests/Repositories/GetFileRepositoryTests.cs]: Fixed

File: DigitalOffice/Tests/FileServiceTests/Repositories/GetFileRepositoryTests.cs (hunk 4 above)
Comment (lalagvana) [reply DigitalOffice/Tests/FileServiceTests/Repositories/GetFileRepositoryTests.cs]: Fixed

File: DigitalOffice/Services/FileService/Repositories/Interfaces/IFileRepository.cs (hunk 3 above)
Comment (lalagvana) [reply DigitalOffice/Services/FileService/Repositories/Interfaces/IFileRepository.cs]: Fixed

File: DigitalOffice/Tests/FileServiceTests/Commands/GetFileCommandTests.cs (hunk 2 above)
Comment (lalagvana) [reply DigitalOffice/Tests/FileServiceTests/Commands/GetFileCommandTests.cs]: Fixed

File: DigitalOffice/Tests/FileServiceTests/Commands/GetFileCommandTests.cs (hunk 1 above)
Comment (lalagvana) [reply DigitalOffice/Tests/FileServiceTests/Commands/GetFileCommandTests.cs]: Fixed

**Review by denis-kriv — APPROVED** [review]
I think, that we have to create only one tests class for every commands, repositories...

File: DigitalOffice/Services/FileService/Controllers/FileController.cs [hunk 6]
```diff
@@ -16,5 +17,12 @@ public class FileController : ControllerBase
         {
             return command.Execute(request);
         }
+
+        [Route("getFile")]
```
Comment (KostinP) [inline DigitalOffice/Services/FileService/Controllers/FileController.cs]: getFileById?

Comment (lalagvana) [reply DigitalOffice/Services/FileService/Controllers/FileController.cs]: Well, agreed. Done.

File: DigitalOffice/Services/FileService/Startup.cs [hunk 7]
```diff
@@ -60,8 +61,8 @@ private void ConfigureValidators(IServiceCollection services)
         {
             services.AddTransient<IValidator<FileCreateRequest>, NewFileValidator>();
         }
-        
-        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
```
Comment (KostinP) [inline DigitalOffice/Services/FileService/Startup.cs:64]: By the way, this is an extra variable in all services. Are you suggesting to remove it?

File: DigitalOffice/Tests/FileServiceTests/Commands/GetFileByIdCommandTests.cs [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    public class GetFileByIdCommandTests
+    {
+        private IGetFileByIdCommand command;
+        private Mock<IFileRepository> repositoryMock;
+        private Mock<IMapper<DbFile, File>> mapperMock;
+
+        private DbFile file;
+        private Guid fileId;
+
+        private readonly string Name = "File";
```
Comment (KostinP) [inline DigitalOffice/Tests/FileServiceTests/Commands/GetFileByIdCommandTests.cs]: Fields are named with a lowercase letter.

Comment (KostinP) [reply DigitalOffice/Tests/FileServiceTests/Commands/GetFileByIdCommandTests.cs]: Not sure if this should initialize here.
https://github.com/LT-Students/DigitalOffice/pull/11#discussion_r460458483

File: DigitalOffice/Services/FileService/Startup.cs (hunk 7 above)
Comment (lalagvana) [reply DigitalOffice/Services/FileService/Startup.cs:64]: We can discuss it on our conversation this evening :)

Comment (lalagvana) [reply DigitalOffice/Services/FileService/Startup.cs:64]: Cause I can't definately say if it's okay to remove it everywhere.

File: DigitalOffice/Tests/FileServiceTests/Commands/GetFileByIdCommandTests.cs (hunk 8 above)
Comment (lalagvana) [reply DigitalOffice/Tests/FileServiceTests/Commands/GetFileByIdCommandTests.cs]: Fixed. Moved it exactly to initializing DbFile

**Review by KostinP — APPROVED** [review]

File: DigitalOffice/Services/FileService/Commands/GetFileByIdCommand.cs [hunk 9]
```diff
@@ -0,0 +1,29 @@
+﻿using LT.DigitalOffice.FileService.Commands.Interfaces;
+using LT.DigitalOffice.FileService.Database.Entities;
+using LT.DigitalOffice.FileService.Mappers.Interfaces;
+using LT.DigitalOffice.FileService.Models;
+using LT.DigitalOffice.FileService.Repositories.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using System;
```
Comment (djino97) [inline DigitalOffice/Services/FileService/Commands/GetFileByIdCommand.cs]: First connect system namespaces and than namespaces created in the our project

File: DigitalOffice/Tests/FileServiceTests/Commands/GetFileByIdCommandTests.cs [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+namespace LT.DigitalOffice.FileServiceUnitTests.Commands
+{
+    public class GetFileByIdCommandTests
+    {
+        private IGetFileByIdCommand command;
+        private Mock<IFileRepository> repositoryMock;
+        private Mock<IMapper<DbFile, File>> mapperMock;
+
+        private DbFile file;
```
Comment (djino97) [inline DigitalOffice/Tests/FileServiceTests/Commands/GetFileByIdCommandTests.cs]: We do not separate fields from each other

File: DigitalOffice/Tests/FileServiceTests/Repositories/AddNewFileRepositoryTests.cs [hunk 11]
```diff
@@ -41,15 +41,15 @@ public void SuccessfulAddNewFileToDatabaseTest()
...
         }
-        
+
         [Test]
         public void FailedAddNewFileFileAlreadyExistsTest()
         {
-            repository.AddNewFile(newFile);            
+            repository.AddNewFile(newFile);
             Assert.Throws<ArgumentException>(() => repository.AddNewFile(newFile));
             Assert.NotNull(dbContext.Files.Find(newFile.Id));
```
Comment (djino97) [inline DigitalOffice/Tests/FileServiceTests/Repositories/AddNewFileRepositoryTests.cs]: Add empty line

File: DigitalOffice/Services/FileService/Commands/GetFileByIdCommand.cs (hunk 9 above)
Comment (lalagvana) [reply DigitalOffice/Services/FileService/Commands/GetFileByIdCommand.cs]: Fixed

File: DigitalOffice/Tests/FileServiceTests/Commands/GetFileByIdCommandTests.cs (hunk 10 above)
Comment (lalagvana) [reply DigitalOffice/Tests/FileServiceTests/Commands/GetFileByIdCommandTests.cs]: Fixed

File: DigitalOffice/Tests/FileServiceTests/Repositories/AddNewFileRepositoryTests.cs (hunk 11 above)
Comment (lalagvana) [reply DigitalOffice/Tests/FileServiceTests/Repositories/AddNewFileRepositoryTests.cs]: Fixed

**Review by djino97 — APPROVED** [review]

File: DigitalOffice/Services/FileService/Commands/Interfaces/IGetFileByIdCommand.cs [hunk 12]
```diff
@@ -0,0 +1,10 @@
+﻿using LT.DigitalOffice.FileService.Models;
+using System;
+
+namespace LT.DigitalOffice.FileService.Commands.Interfaces
+{
+    public interface IGetFileByIdCommand
```
Comment (MihailKulikov) [inline DigitalOffice/Services/FileService/Commands/Interfaces/IGetFileByIdCommand.cs:6]: XML documentation

File: DigitalOffice/Services/FileService/Repositories/FileRepository.cs [hunk 13]
```diff
@@ -19,8 +20,20 @@ public Guid AddNewFile(DbFile file)
...
             dbContext.Files.Add(file);
             dbContext.SaveChanges();
-            
+
             return file.Id;
         }
+
+        public DbFile GetFileById(Guid fileId)
+        {
+            var file = dbContext.Files.FirstOrDefault(file => file.Id == fileId);
```
Comment (MihailKulikov) [inline DigitalOffice/Services/FileService/Repositories/FileRepository.cs]: parameter 'file' hides outer local variable with the same name. Maybe name of variable should be dbFile(it gives more useful information)

Comment (lalagvana) [reply DigitalOffice/Services/FileService/Repositories/FileRepository.cs]: Fixed

File: DigitalOffice/Services/FileService/Commands/Interfaces/IGetFileByIdCommand.cs (hunk 12 above)
Comment (lalagvana) [reply DigitalOffice/Services/FileService/Commands/Interfaces/IGetFileByIdCommand.cs:6]: What do you mean?

Comment (KostinP) [reply DigitalOffice/Services/FileService/Commands/Interfaces/IGetFileByIdCommand.cs:6]: Most likely he offers you to add documentation to the interface.

**Review by seasmail — APPROVED** [review]

**Pull request merged**